cocotb/share/lib/simulator/simulatormodule.c

```diff
@@ -654,7 +653,24 @@ static PyObject *set_signal_val_str(PyObject *self, PyObject *args)
         return NULL;
     }
 
-    gpi_set_signal_value_str(hdl, binstr);
+    gpi_set_signal_value_binstr(hdl, binstr);
+    res = Py_BuildValue("s", "OK!");
+
+    return res;
+}
+
+static PyObject *set_signal_val_str(PyObject *self, PyObject *args)
+{
+    COCOTB_UNUSED(self);
+    gpi_sim_hdl hdl;
+    const char *str;
+    PyObject *res;
+
+    if (!PyArg_ParseTuple(args, "O&s", gpi_sim_hdl_converter, &hdl, &str)) {
```

Comment:
This null-terminates strings - are you sure you want that? Otherwise, you want `y#` here I think.

Reply:
Edit: Looks like you do want embedded nulls - but `y` would be more appropriate, since HDL doesn't know about unicode.

Reply:
I made this a separate commit for you to ok.